frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/LibraryOptionsEditor/LibraryOptionsEditor.tsx

```diff
@@ -1,70 +1,69 @@
-import { useOptionListEditorTexts } from '../../../../../EditOptions/OptionTabs/hooks';
-import classes from './LibraryOptionsEditor.module.css';
+import React, { createRef } from 'react';
 import { useTranslation } from 'react-i18next';
+import { StudioCodeListEditor, StudioModal, StudioAlert } from '@studio/components';
 import type { CodeListEditorTexts } from '@studio/components';
-import { StudioAlert, StudioCodeListEditor, StudioModal, StudioProperty } from '@studio/components';
 import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from 'app-shared/hooks/mutations';
+import { useOptionListEditorTexts } from '../../../hooks';
+import { OptionListButtons } from '../OptionListButtons';
+import { OptionListLabels } from '../OptionListLabels';
+import { hasOptionListChanged } from '../../../utils/optionsUtils';
+import type { OptionListEditorProps } from '../OptionListEditor';
+import classes from './LibraryOptionsEditor.module.css';
 
 type LibraryOptionsEditorProps = {
-  optionsId: string;
   optionsList: Option[];
-};
+  handleDelete: () => void;
+} & Pick<OptionListEditorProps, 'component'>;
 
 export function LibraryOptionsEditor({
-  optionsId,
+  component,
   optionsList,
+  handleDelete,
 }: LibraryOptionsEditorProps): React.ReactNode {
   const { t } = useTranslation();
   const { org, app } = useStudioEnvironmentParams();
   const { doReloadPreview } = usePreviewContext();
   const { mutate: updateOptionList } = useUpdateOptionListMutation(org, app);
-  const [localOptionList, setLocalOptionList] = useState<Option[]>(optionsList);
   const editorTexts: CodeListEditorTexts = useOptionListEditorTexts();
   const modalRef = createRef<HTMLDialogElement>();
 
-  const optionListHasChanged = (options: Option[]): boolean =>
-    JSON.stringify(options) !== JSON.stringify(localOptionList);
-
   const handleBlurAny = (options: Option[]) => {
-    if (optionListHasChanged(options)) {
-      updateOptionList({ optionListId: optionsId, optionsList: options });
-      setLocalOptionList(options);
+    if (hasOptionListChanged(component.options, options)) {
+      updateOptionList({ optionListId: component.optionsId, optionsList: options });
       doReloadPreview();
     }
   };
 
-  const handleClose = () => {
-    modalRef.current?.close();
+  const handleClick = () => {
+    modalRef.current?.showModal();
   };
 
   return (
     <>
-      <StudioProperty.Button
-        value={optionsId}
-        title={t('ux_editor.options.option_edit_text')}
-        property={t('ux_editor.modal_properties_code_list_button_title_library')}
-        onClick={() => modalRef.current.showModal()}
+      <OptionListLabels component={component} optionsList={optionsList} />
+      <OptionListButtons
+        handleClick={handleClick}
+        component={component}
+        optionsList={optionsList}
```

Comment:
I think we do not need to pass these two either? 
Then we can actually remove the component as a prop from the `LibraryOptionsEditor` all together and only pass the optionsId.